Web.md

```diff
@@ -124,6 +124,81 @@ MIME type mismatch or `opaque` response types
+          (dataCopy, 0,
+          value.[[BufferObject]].[[ArrayBufferData]], 0, size).
+       3. Set `serialized` to { [[Type]]: "WebAssembly.Memory",
+          [[ArrayBufferData]]: dataCopy, [[ArrayBufferByteLength]]: size }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if `serialized`.[[Type]] is "WebAssembly.Memory", then:
+  1. If `targetRealm`'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     `serialized`.[[AgentCluster]], then throw a
+     ["DataCloneError" DomException](https://heycam.github.io/webidl/#dfn-DOMException).
+  2. Otherwise, set `value` to a new WebAssembly.Memory object in `targetRealm`
+     whose [[ArrayBufferData]] internal slot value is
+     `serialized`.[[ArrayBufferData]]
+     and whose [[ArrayBufferByteLength]] internal
+     slot value is `serialized`.[[ArrayBufferByteLength]].
+
+
+### Structured Clone of a `WebAssembly.Module`
+
+A `WebAssembly.Module` is under some circumstances a
+[cloneable object](https://html.spec.whatwg.org/multipage/infrastructure.html#cloneable-objects), and will impact the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if `value` has an [[WebAssembly.Module]] internal slot, then:
+  1. If `forStorage` is true, implementations MAY chose to exclude
+     "insecure origins", in which case they should throw a
+     ["DataCloneError" DomException](https://heycam.github.io/webidl/#datacloneerror).
+  2. Set `serialized` to { [[Type]]: "WebAssembly.Module",
+     [[Module]]: value.[[Module]],
+     [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if `serialized`.[[Type]] is "WebAssembly.Module", then:
+  1. If `targetRealm`'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     `serialized`.[[AgentCluster]], then throw a
+     ["DataCloneError" DomException](https://heycam.github.io/webidl/#dfn-DOMException).
+  2. Otherwise, set `value` to a new WebAssembly.Module object in `targetRealm`
+     whose [[Module]] internal slot value is `serialized`.[[Module]].
+
+The semantics of a structured clone is as-if the binary source, from which the
+`WebAssembly.Module` was compiled, were cloned and recompiled into the target realm.
+Engines should attempt to share/reuse internal compiled code when performing
+a structured clone although, in corner cases like CPU upgrade or browser
```

Comment:
Nit: comma should be maybe "when performing structured serialization, although in corner cases..."